src/mbgl/layout/symbol_layout.cpp

```diff
@@ -254,7 +254,7 @@ void SymbolLayout::addFeature(const SymbolFeature& feature,
                                                   ? SymbolPlacementType::Point
                                                   : layout.get<SymbolPlacement>();
     const float textRepeatDistance = symbolSpacing / 2;
-    IndexedSubfeature indexedFeature = {feature.index, sourceLayerName, layers.at(0)->getID(), symbolInstances.size()};
+    IndexedSubfeature indexedFeature = {feature.index, sourceLayerName, layerIDs.at(0), symbolInstances.size()};
```

Comment:
That invariant should hold: the layer IDs are constructed from the layer group, and if there's no group, no `SymbolLayout` will be created. However, just in case it doesn't, I used `at(0)` here rather than `[0]` to get an exception rather than invalid memory access.